### PR TITLE
Exclude schema registry and HTTP proxy config from 22.2.X deployment

### PR DESCRIPTION
**Pull request by RafalKorepta**
As version 22.2.X does not provide basic auth configuration option in both schema registry and panda proxy the client configuration for both services needs SASL credentails to be explicitly set in Redpanda configuration. That Redpanda configuration is generated in helm and sotred as config map which should not hold any credentails.

Newer version can take HTTP basic auth header to impersonate user and authenticate against Redpanda Kafka listener.

Additionally HTTP proxy and schema registry tests now are run when SASL is enabled

### CODE REVIEW

**Comment by RafalKorepta** [comment]
Tests are blocked by https://github.com/redpanda-data/helm-charts/pull/547

**Comment by RafalKorepta** [comment]
Test 05 was adjusted to use not default settings https://github.com/redpanda-data/helm-charts/pull/550. I hope upgrade tests will pass now.

File: charts/redpanda/templates/_helpers.tpl [hunk 1]
```diff
@@ -538,6 +538,10 @@ than 1 core.
 {{- toJson (dict "bool" (or (not (eq .Values.image.repository "docker.redpanda.com/redpandadata/redpanda")) (include "redpanda.semver" . | semverCompare ">=22.2.10-0,<22.3"))) -}}
 {{- end -}}
 
+{{- define "redpanda-22-2-x-without-sasl" -}}
+{{- toJson (dict "bool" (or (not (or (include "sasl-enabled" . | fromJson).bool .Values.listeners.kafka.authenticationMethod)) (include "redpanda-atleast-22-3-0" . | fromJson).bool)) -}}
```
Comment (joejulian) [inline charts/redpanda/templates/_helpers.tpl:542]: If this is accurate, I'd suggest this. The nested logic is way to hard to parse in my head.

```suggestion
{{- $result :=  (include "redpanda-atleast-22-3-0" . | fromJson).bool -}}
{{- if or (include "sasl-enabled" . | fromJson).bool .Values.listeners.kafka.authenticationMethod)) -}}
{{-   $result := false -}}
{{- end -}}
{{- toJson (dict "bool" $result -}}
```

File: charts/redpanda/templates/tests/test-schemaregistry-internal-tls-status.yaml [hunk 2]
```diff
@@ -41,40 +42,94 @@ spec:
...
+
+          curl -svm3 --fail --retry "120" --retry-max-time "120" --retry-all-errors \
+          -X POST -H "Content-Type: application/vnd.schemaregistry.v1+json" \
+          -d '{"schema": "{\"type\":\"record\",\"name\":\"sensor_sample\",\"fields\":[{\"name\":\"timestamp\",\"type\":\"long\",\"logicalType\":\"timestamp-millis\"},{\"name\":\"identifier\",\"type\":\"string\",\"logicalType\":\"uuid\"},{\"name\":\"value\",\"type\":\"long\"}]}"}' \
+          {{- if or (include "sasl-enabled" .|fromJson).bool .Values.listeners.schemaRegistry.authenticationMethod }}
+          -u $USERNAME:$PASSWORD \
+          {{- end }}
+          {{- if $cert.caEnabled }}
+          --cacert /etc/tls/certs/{{ $service.tls.cert }}/ca.crt \
+          {{- end }}
```
Comment (joejulian) [inline charts/redpanda/templates/tests/test-schemaregistry-internal-tls-status.yaml:70]: This is repeated for each command. What do you think about putting it in a function?

```
schemaCurl () {
          {{- if or (include "sasl-enabled" .|fromJson).bool .Values.listeners.schemaRegistry.authenticationMethod }}
          -u $USERNAME:$PASSWORD \
          {{- end }}
          {{- if $cert.caEnabled }}
          --cacert /etc/tls/certs/{{ $service.tls.cert }}/ca.crt \
          {{- end }} $1
}
```

```
schemaCurl https://{{ include "redpanda.internal.domain" . }}:{{ .Values.listeners.schemaRegistry.port }}/subjects/sensor-value/versions
```

Comment (RafalKorepta) [reply charts/redpanda/templates/tests/test-schemaregistry-internal-tls-status.yaml:70]: Great suggestion

File: charts/redpanda/templates/tests/test-schemaregistry-status.yaml [hunk 3]
```diff
@@ -38,14 +39,64 @@ spec:
...
+          {{- end }}
+          http://{{ include "redpanda.internal.domain" . }}:{{ .Values.listeners.schemaRegistry.port }}/schemas/types
+
+          curl -svm3 --fail --retry "120" --retry-max-time "120" --retry-all-errors \
+          -X POST -H "Content-Type: application/vnd.schemaregistry.v1+json" \
+          -d '{"schema": "{\"type\":\"record\",\"name\":\"sensor_sample\",\"fields\":[{\"name\":\"timestamp\",\"type\":\"long\",\"logicalType\":\"timestamp-millis\"},{\"name\":\"identifier\",\"type\":\"string\",\"logicalType\":\"uuid\"},{\"name\":\"value\",\"type\":\"long\"}]}"}' \
+          {{- if or (include "sasl-enabled" .|fromJson).bool .Values.listeners.schemaRegistry.authenticationMethod }}
+          -u $USERNAME:$PASSWORD \
+          {{- end }}
+          http://{{ include "redpanda.internal.domain" . }}:{{ .Values.listeners.schemaRegistry.port }}/subjects/sensor-value/versions
```
Comment (joejulian) [inline charts/redpanda/templates/tests/test-schemaregistry-status.yaml:62]: similar to above, could this be made a little more DRY?

File: charts/redpanda/templates/_helpers.tpl (hunk 1 above)
Comment (RafalKorepta) [reply charts/redpanda/templates/_helpers.tpl:542]: I'm not that proficient in templating langue. If I could code this in go :)

Comment included into next push

**Review by joejulian — COMMENTED** [review]
The only change I'd like to see for sure is the breaking up of the compound logic. The other suggestions are just nice-to-haves.

File: charts/redpanda/templates/_helpers.tpl (hunk 1 above)
Comment (joejulian) [reply charts/redpanda/templates/_helpers.tpl:542]: It IS go! :grin:

**Review by joejulian — APPROVED** [review]
